qa/integration-tests/src/test/java/io/zeebe/broker/it/clustering/BrokerLeaderChangeTest.java

```diff
@@ -95,62 +92,40 @@ public void shouldChangeLeaderAfterLeaderDies() {
   }
 
   class JobCompleter {
-
-    private final AtomicBoolean isJobCompleted = new AtomicBoolean(false);
     private final JobWorker jobSubscription;
-    private final TopicSubscription topicSubscription;
+    private final CountDownLatch latch = new CountDownLatch(1);
 
     JobCompleter(final JobEvent jobEvent) {
-      final long eventKey = jobEvent.getMetadata().getKey();
+      final long eventKey = jobEvent.getKey();
 
       jobSubscription =
-          doRepeatedly(
-                  () ->
-                      clientRule
-                          .getJobClient()
-                          .newWorker()
-                          .jobType(JOB_TYPE)
-                          .handler(
-                              (client, job) -> {
-                                if (job.getMetadata().getKey() == eventKey) {
-                                  client.newCompleteCommand(job).payload(NULL_PAYLOAD).send();
-                                }
-                              })
-                          .open())
-              .until(Objects::nonNull, "Failed to open job subscription for job completion");
-
-      topicSubscription =
-          doRepeatedly(
-                  () ->
-                      clientRule
-                          .getClient()
-                          .newSubscription()
-                          .name("jobObserver")
-                          .jobEventHandler(
-                              e -> {
-                                if (JOB_TYPE.equals(e.getType())
-                                    && e.getState() == JobState.COMPLETED) {
-                                  isJobCompleted.set(true);
-                                }
-                              })
-                          .startAtHead()
-                          .forcedStart()
-                          .open())
-              .until(Objects::nonNull, "Failed to open topic subscription for job completion");
+          clientRule
+              .getJobClient()
+              .newWorker()
+              .jobType(JOB_TYPE)
+              .handler(
+                  (client, job) -> {
+                    if (job.getKey() == eventKey) {
+                      client.newCompleteCommand(job.getKey()).payload(NULL_PAYLOAD).send();
+                      latch.countDown();
+                    }
+                  })
+              .open();
     }
 
     void waitForJobCompletion() {
-      waitUntil(isJobCompleted::get, 100, "Failed to wait for job completion");
+      try {
+        latch.await();
```

Comment:
This will block forever, better specify a timeout and check the return value. Or we could think about using the new activate/complete API without relying on job workers.